rewrite-maven/src/main/java/org/openrewrite/maven/RemoveDuplicateDependencies.java

```diff
@@ -0,0 +1,147 @@
+
+    @Override
+    public String getDisplayName() {
+        return "Remove duplicate Maven dependencies";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Removes duplicated dependencies in the `<dependencies>` and `<dependencyManagement>` sections of the `pom.xml`.";
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(2);
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new RemoveDuplicateDependenciesVisitor();
+    }
+
+    @Override
+    protected MavenIsoVisitor<ExecutionContext> getApplicableTest() {
+        return new MavenIsoVisitor<ExecutionContext>() {
+            @Override
+            public Xml.Document visitDocument(Xml.Document document, ExecutionContext executionContext) {
+                Xml.Tag root = document.getRoot();
+                if (root.getChild("dependencies").isPresent() || root.getChild("dependencyManagement").isPresent()) {
+                    return SearchResult.found(document);
+                }
+                return document;
+            }
+        };
+    }
+
+    private static class RemoveDuplicateDependenciesVisitor extends MavenIsoVisitor<ExecutionContext> {
+        private static final XPathMatcher DEPENDENCIES_MATCHER = new XPathMatcher("/project/dependencies");
+        private static final XPathMatcher MANAGED_DEPENDENCIES_MATCHER = new XPathMatcher("/project/dependencyManagement/dependencies");
+
+        @Override
+        public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext ctx) {
+            if (isDependenciesTag()) {
+                ctx.putMessage("dependencies", new HashMap<DependencyKey, Xml.Tag>());
+            } else if (isManagedDependenciesTag()) {
+                ctx.putMessage("managedDependencies", new HashMap<DependencyKey, Xml.Tag>());
+            } else if (isDependencyTag()) {
+                Map<DependencyKey, Xml.Tag> dependencies = ctx.getMessage("dependencies");
+                ResolvedDependency resolvedDependency = findDependency(tag);
+                Xml.Tag existing = dependencies.putIfAbsent(DependencyKey.from(resolvedDependency), tag);
+                if (existing != null && existing != tag) {
+                    maybeUpdateModel();
+                    return null;
+                }
+            } else if (isManagedDependencyTag()) {
+                Map<DependencyKey, Xml.Tag> dependencies = ctx.getMessage("managedDependencies");
+                ResolvedManagedDependency resolvedDependency = findManagedDependency(tag);
+                Xml.Tag existing = dependencies.putIfAbsent(DependencyKey.from(resolvedDependency), tag);
+                if (existing != null && existing != tag) {
+                    maybeUpdateModel();
+                    return null;
+                }
+            }
+            return super.visitTag(tag, ctx);
+        }
+
+        private boolean isDependenciesTag() {
```

Comment:
The `isDependenciesTag()`, `isManagedDependenciesTag()` and `findDependency()` are all in the MavenVisitor, so you can remove these three, in favor of using those.

Reply:
I think I added copies here, because I need them to work slightly differently. But I will check again and if I still need them, I will rename them and / or add some comment to this effect.

Reply:
`isDependenciesTag()` and `isManagedDependenciesTag()` I couldn't find and `findDependency(Tag)` doesn't quite do what I want, since it also considers the version, which I for this use case don't want to do (should also be considered a duplicate).